subsys/debug/coredump/coredump_mw_backend_intel_adsp.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/debug/coredump.h>
+#include "coredump_internal.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <adsp_memory.h>
+#include <adsp_debug_window.h>
+LOG_MODULE_REGISTER(coredump_error, CONFIG_KERNEL_LOG_LEVEL);
+
+static int error;
+
+static void coredump_mem_window_backend_start(void)
+{
+	/* Reset error */
+	error = 0;
+	ADSP_DW->descs[1].type = ADSP_DW_SLOT_TELEMETRY;
+
+	while (LOG_PROCESS()) {
+		;
+	}
+
+	LOG_PANIC();
+	LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_BEGIN_STR);
+}
+
+static void coredump_mem_window_backend_end(void)
+{
+	if (error != 0) {
+		LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_ERROR_STR);
+	}
+
+	LOG_ERR(COREDUMP_PREFIX_STR COREDUMP_END_STR);
+}
+
+static void coredump_mem_window_backend_buffer_output(uint8_t *buf, size_t buflen)
+{
+	uint8_t *mem_window_sink = (uint8_t *)(ADSP_DW->slots[1]);
```

Comment:
But should we write some header at start of the slot to make it possible to discover this specific layout for the telemetry window is used?

We know older IPC4 binaries will have "telemetry_data->separator_1 = 0x0000C0DE;" at start, so maybe just add a 32bit constant value (different from C0DE) at start here. Then it's easy to detect conflicts.

Reply:
I agree. Life would be easier that way but it is going to be temporary solution at least. Should 0x0DEC0DEB be better?